src/datasources.js

```diff
@@ -225,11 +246,50 @@ define([
         return widget;
     }
 
+    function fixtureDataSourceWidget(mug, options, labelText) {
+        var widget = widgets.dropdown(mug, options),
+            getUIElement = widgets.util.getUIElement,
+            super_getValue = widget.getValue,
+            super_setValue = widget.setValue,
+            currentValue = null;
+
+        getDataSources("fixture", generateFixtureQueries(widget));
+
+        widget.getUIElement = function () {
+            var query = getUIElement(widget.input, labelText);
+            return $("<div></div>").append(query);
+        };
+
+        function local_getValue() {
+            currentValue.src = super_getValue();
+            return currentValue;
+        }
+
+        function local_setValue(val) {
+            currentValue = val;
+            super_setValue(val.src || "");
+        }
+
+        widget.getValue = local_getValue;
+        widget.setValue = local_setValue;
+
+        return widget;
+    }
+
+    $.vellum.plugin("datasources", { }, {
```

Comment:
We probably want to change the `init` function so it's called the same way as other plugins rather than the one-off way it's done by core.js right now.